### PR TITLE
refactor: minor refactoring + optimization

### PR DESCRIPTION
**Pull request by SohaibRaza**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [x] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 1e8c287</samp>

This file improves the `Button` component's performance and fixes a minor bug. It uses `useCallback` to avoid unnecessary re-rendering and corrects a misspelled variable.

### <samp>🤖 Generated by Copilot at 1e8c287</samp>

*  Wrap `fixTwoCNChar` and `handleClick` functions with `useCallback` and update their dependencies arrays to optimize performance and avoid stale closures ([link](https://github.com/ant-design/ant-design/pull/42037/files?diff=unified&w=0#diff-463cd38ab465e1f3b90463a403d8e8399f926361436eaa9adb8a76a8bf9d4140L137-R137), [link](https://github.com/ant-design/ant-design/pull/42037/files?diff=unified&w=0#diff-463cd38ab465e1f3b90463a403d8e8399f926361436eaa9adb8a76a8bf9d4140L150-R150), [link](https://github.com/ant-design/ant-design/pull/42037/files?diff=unified&w=0#diff-463cd38ab465e1f3b90463a403d8e8399f926361436eaa9adb8a76a8bf9d4140L176-R187))
* Fix typo and naming convention of `sizeFullName` variable in `components/button/button.tsx` ([link](https://github.com/ant-design/ant-design/pull/42037/files?diff=unified&w=0#diff-463cd38ab465e1f3b90463a403d8e8399f926361436eaa9adb8a76a8bf9d4140L202-R206))

### CODE REVIEW

**Comment by yoyo837** [comment]
The need for this change doesn't seem obvious, what do you think?

**Comment by SohaibRaza** [comment]
> The need for this change doesn't seem obvious, what do you think?

While it may only be a minor refactoring and code optimization, every little improvement we make can have a significant impact on the overall quality and performance of the code. By continuously making small improvements, we can collectively make a substantial difference in the long run. As the **Boy Scout Rule** suggests 'Always leave the code better than you found it. '

**Comment by afc163** [comment]
Conflicting

File: components/button/button.tsx [hunk 1]
```diff
@@ -178,15 +178,18 @@ const InternalButton: React.ForwardRefRenderFunction<
...
+    (e: React.MouseEvent<HTMLButtonElement | HTMLAnchorElement, MouseEvent>) => {
+      const { onClick } = props;
+      // FIXME: https://github.com/ant-design/ant-design/issues/30207
+      if (innerLoading || mergedDisabled) {
+        e.preventDefault();
+        return;
+      }
+      (onClick as React.MouseEventHandler<HTMLButtonElement | HTMLAnchorElement>)?.(e);
+    },
+    [props, innerLoading, mergedDisabled],
```
Comment (afc163) [inline components/button/button.tsx:191]: `props` would be truthy all the time.

**Comment by afc163** [comment]
ping

**Comment by afc163** [comment]
ping~ @SohaibRaza

**Review by afc163 — APPROVED** [review]

**Pull request merged**